Connecting_MonteCarlo_to_ModernAI/N2_2D_Ising_Model.ipynb

```diff
@@ -0,0 +1,1233 @@
+        "\n",
+        "---\n",
+        "\n",
+        "0. Initialize the system with the following variables:\n",
+        "    - $T$: temperature of the \"bath\" which will anneal the system\n",
+        "    - $N$: the size of the system\n",
+        "    - $s$: the state of each particle in the system\n",
+        "    - $J$: the coupling between particles in the system\n",
+        "    - $H$: the applied magnetic field\n",
+        "1. Select an atom in the system\n",
+        "2. Calculate the energy $e_i$ of that atom \n",
+        "3. Determine if changing the spin of the atom would decrease the energy\n",
+        "4. If it decreases the energy **OR** if the probability of the state in the Boltzmann distribution is greater than a sample from a uniform probability distribution, then change the spin state of the atom.\n",
+        "5. Continue with steps 1-4 until the stopping criteria is met.\n",
+        "6. Return to step 0 and re-initialize the system with different variables as desired.\n",
+        "---\n",
+        "Let's go through these steps in code:"
+      ],
+      "metadata": {
+        "id": "CKoPXdP1sZ5W"
+      }
+    },
+    {
+      "cell_type": "markdown",
+      "source": [
+        "**Step 0: Initialize the lattice and the simulation**\n",
+        "\n",
+        "I want a square, 10x10 lattice (N = 100).  By calling the `np.random.uniform()` function, I know I will get an uniform distribution of values between 0 and 1"
+      ],
+      "metadata": {
+        "id": "bGQwjA0XAdnh"
+      }
+    },
+    {
+      "cell_type": "code",
+      "source": [
+        "lattice = np.random.uniform(size=(10,10))"
+      ],
+      "metadata": {
+        "id": "iskEhLNBAlwm"
+      },
+      "execution_count": null,
+      "outputs": []
+    },
+    {
+      "cell_type": "markdown",
+      "source": [
+        "Next, I want to prepare the system into spin up and spin down"
+      ],
+      "metadata": {
+        "id": "7-JJJvoyAn9_"
+      }
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "id": "AvTQs_NkPdgj"
+      },
+      "outputs": [],
+      "source": [
+        "lattice[lattice>0.5]=1 # any values greater than 0.5, I set equal to 1\n",
+        "lattice[lattice !=1]=-1 # anything I don't change before, I make equal to -1"
+      ]
```

Comment:
It may be easier (to do and to explain) to just use `np.random.choice([-1,1], size=(10,10))` which just randomly chooses -1 or 1 for all points.

Reply:
Thank you for your comment.  While I agree that this function is a more elegant solution, it is also hyper-specific to the Python programming language and obscures the idea of preparing a system for simulation.  Other coding languages will still require the student to randomly draw from a distribution and then assign a value.  I have updated the plotting code to use this function, but left the explanation as is.